packages/core/src/browser/tree/fuzzy-search.ts

```diff
@@ -14,123 +14,25 @@
  * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
  ********************************************************************************/
 
-import * as fuzzy from 'fuzzy';
 import { injectable } from 'inversify';
-
-@injectable()
-export class FuzzySearch {
-
-    private static readonly PRE = '\x01';
-    private static readonly POST = '\x02';
-
-    /**
-     * Filters the input and returns with an array that contains all items that match the pattern.
-     */
-    async filter<T>(input: FuzzySearch.Input<T>): Promise<FuzzySearch.Match<T>[]> {
-        return fuzzy.filter(input.pattern, input.items.slice(), {
-            pre: FuzzySearch.PRE,
-            post: FuzzySearch.POST,
-            extract: input.transform
-        }).sort(this.sortResults.bind(this)).map(this.mapResult.bind(this));
-    }
-
-    protected sortResults<T>(left: fuzzy.FilterResult<T>, right: fuzzy.FilterResult<T>): number {
-        return left.index - right.index;
-    }
-
-    protected mapResult<T>(result: fuzzy.FilterResult<T>): FuzzySearch.Match<T> {
-        return {
-            item: result.original,
-            ranges: this.mapRanges(result.string)
-        };
-    }
-
-    protected mapRanges(input: string): ReadonlyArray<FuzzySearch.Range> {
-        const copy = input.split('').filter(s => s !== '');
-        const ranges: FuzzySearch.Range[] = [];
-        const validate = (pre: number, post: number) => {
-            if (preIndex > postIndex || (preIndex === -1) !== (postIndex === -1)) {
-                throw new Error(`Error when trying to map ranges. Escaped string was: '${input}. [${[...input].join('|')}]'`);
-            }
-        };
-        let preIndex = copy.indexOf(FuzzySearch.PRE);
-        let postIndex = copy.indexOf(FuzzySearch.POST);
-        validate(preIndex, postIndex);
-        while (preIndex !== -1 && postIndex !== -1) {
-            ranges.push({
-                offset: preIndex,
-                length: postIndex - preIndex - 1
-            });
-            copy.splice(postIndex, 1);
-            copy.splice(preIndex, 1);
-            preIndex = copy.indexOf(FuzzySearch.PRE);
-            postIndex = copy.indexOf(FuzzySearch.POST);
-        }
-        if (ranges.length === 0) {
-            throw new Error(`Unexpected zero ranges for match-string: ${input}.`);
-        }
-        return ranges;
-    }
-
-}
+import { FuzzySearch as CommonFuzzySearch } from '../../common/fuzzy-search';
 
 /**
- * Fuzzy searcher.
+ * @deprecated import from `packages/core/src/common/fuzzy-search` instead.
  */
+@injectable()
+export class FuzzySearch extends CommonFuzzySearch { }
```

Comment:
added a comment to the commit by the mistake: https://github.com/eclipse-theia/theia/commit/fbdd35a0227706424c15fe9b3b3c51e63110016b#r38021579

Reply:
OK, nevermind.